pkg/resource/v1/certconfig/current.go

```diff
@@ -2,8 +2,67 @@ package certconfig
 
 import (
 	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/cluster-operator/pkg/resource/v1/certconfig/key"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/randomkeytpr"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// GetCurrentState takes observed custom object as an input and based on that
+// information looks for current state of cluster certconfigs and returns them.
+// Return value is of type []*v1alpha1.CertConfig.
 func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
+	clusterGuestConfig, err := r.toClusterGuestConfigFunc(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "debug", "looking for a list of certconfigs in the Kubernetes API")
+
+	var certConfigs []*v1alpha1.CertConfig
+	{
+		labelSelector := &metav1.LabelSelector{}
+		labelSelector = metav1.AddLabelToSelector(labelSelector, randomkeytpr.ClusterIDLabel, key.ClusterID(*clusterGuestConfig))
```

Comment:
I think you want to use `label` instead of `randomkeytpr`.

Reply:
In the future yes, but for transition period `randomkeytpr` label is the correct one in order to find `CertConfigs` that are created by `kubernetesd`.

Reply:
So shouldn't be that `certificatetpr`?
BTW does it make sense to import a repo for a label? Maybe it makes sense to create something like `LegacyClusterID`?

Reply:
Good point. Added `LegacyClusterID`.